src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs

```diff
@@ -695,5 +697,41 @@ private void PersistOnShutdown()
         }
 
         #endregion
+        
+        public IReadOnlyTrieStore AsReadOnly(IKeyValueStore readOnlyStore)
+        {
+            return new ReadOnlyTrieStore(this, readOnlyStore);
+        }
+        
+        private class ReadOnlyTrieStore : IReadOnlyTrieStore
```

Comment:
why internal class, keep separately, in a separate file
this class is used externally